usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MedicationManagerClient.java

```diff
@@ -0,0 +1,214 @@
+    }
+
+    /**
+     * Deletes all the Medications of the specified pet from database.
+     * @param accessToken The personal access token for the account.
+     * @param owner Username of the owner of the pet.
+     * @param petName Name of the pet.
+     * @return The response code.
+     * @throws ExecutionException When the deletion fails.
+     * @throws InterruptedException When the deletion is interrupted.
+     */
+    public int deleteAllMedications(String accessToken, String owner, String petName)
+            throws ExecutionException,
+            InterruptedException {
+        taskManager = taskManager.resetTaskManager();
+        taskManager.setTaskId(DELETE);
+        StringBuilder response = taskManager.execute(BASE_URL + owner + DASH + petName, accessToken).get();
+        return Integer.parseInt(response.toString());
+    }
+
+    /**
+     * Gets a Medication identified by its pet, date and name.
+     * @param accessToken The personal access token for the account.
+     * @param owner Username of the owner of the pet.
+     * @param petName Name of the pet.
+     * @param date Date the Medication was eaten.
+     * @param name name of the Medication consumed.
+     * @return The MedicationData identified by the data.
+     */
+    public MedicationData getMedicationData(String accessToken, String owner, String petName,
+                                            DateTime date, String name)
+            throws ExecutionException, InterruptedException {
+        taskManager = taskManager.resetTaskManager();
+        taskManager.setTaskId(GET);
+        StringBuilder json = taskManager.execute(BASE_URL + owner + DASH + petName + DASH
+                + date + DASH + name, accessToken).get();
+        return gson.fromJson(json.toString(), MedicationData.class);
+    }
+
+    /**
+     * Gets the data from all the specified Medications from the database identified by its pet.
+     * @param accessToken The personal access token for the account.
+     * @param owner Username of the owner of the pets.
+     * @param petName Name of the pet.
+     * @return The List containing all the Medications from the pet
+     */
+    public List<Medication> getAllMedicationData(String accessToken, String owner, String petName)
+            throws ExecutionException,
+            InterruptedException {
+        taskManager = taskManager.resetTaskManager();
+        taskManager.setTaskId(GET);
+        StringBuilder response = taskManager.execute(BASE_URL + owner + DASH + petName, accessToken).get();
+        List<Medication> medicationList = new ArrayList<>();
+        if (response.length() > 2) {
+            String jsonArray = response.substring(1, response.length() - 1);
+            String[] medications = jsonArray.split(",\\{");
+            medicationList.add(gson.fromJson(medications[0], Medication.class));
+            for (int i = 1; i < medications.length; i++) {
+                medications[i] = "{" + medications[i];
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String "{" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5403182)